crossdock/crossdock_test.go

```diff
@@ -123,12 +113,45 @@ func TestCrossdock(t *testing.T) {
 	}
 }
 
+func wait(t *testing.T) {
+	attempts := 0
+	for {
+		if attempts > 9 {
+			t.Fatalf("could not talk to client in %d attempts", attempts)
+		}
```

Comment:
Nit: Instead of managing count yourself, do:

```go
for attempts := 0; attempts < 10; attempts++ {
  _, err := // make attempt
  if err == nil {
    log.Println("Client is ready, beginning test...")
    return
  }
}
t.Fatal("could not talk to client in 10 attempts")
```